File: cpp/027.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int cur = -1, size = nums.size();
        for (int i = 0; i < size; ++i)
            if (nums[i] != val)
                nums[++cur] = nums[i];
        return cur + 1;
    }
};

File: go/027.go
func removeElement(nums []int, val int) int {
    cur, size := -1, len(nums)
    for i := 0; i < size; i++ {
        if nums[i] != val {
            cur++
            nums[cur] = nums[i]
        }
    }
    return cur + 1
}
